crates/containerd-shim-wasm/src/sandbox/cli.rs

```diff
@@ -151,6 +151,31 @@ fn log_mem() {
     log::info!("Zygote peak memory usage was: peak resident set {rss} kB, peak total {tot} kB");
 }
 
+#[cfg(unix)]
+fn init_zygote_and_logger(debug: bool, config: &Config) -> Result<(), ()> {
+    zygote::Zygote::init();
+    let res = zygote::Zygote::global().run(
+        |(debug, default_log_level, namespace, id, no_setup_logger)| {
+            if !no_setup_logger {
+                if let Err(_) =
+                    containerd_shim::logger::init(debug, &default_log_level, &namespace, &id)
+                {
+                    return Err(());
+                }
+            }
+            Ok(())
+        },
+        (
+            debug,
+            config.default_log_level.clone(),
+            "".to_string(), // namespace is not used in unix
+            "".to_string(), // id is not used in unix
+            config.no_setup_logger,
+        ),
+    );
+    res
+}
```

Comment:
since we terminate the process if this fails, we can simplify it.
`zygote.run` will panic if the closure panics.
`zygote.try_run` is the non panic-ing version of `run`.

```suggestion
fn init_zygote_and_logger(debug: bool, config: &Config) {
    zygote::Zygote::init();
    if config.no_setup_logger {
        return;
    }
    zygote::Zygote::global().run(
        |(debug, default_log_level)| {
            // last two arguments are unused in unix
            containerd_shim::logger::init(debug, &default_log_level, "", "").expect("Failed to initialize logger");
        },
        (debug, config.default_log_level.clone()),
    );
}
```